neo4j/_async_compat/network/_bolt_socket.py

```diff
@@ -605,7 +608,7 @@ def _handshake(cls, s, resolved_address):
     def close_socket(cls, socket_):
         try:
             if isinstance(socket_, BoltSocket):
-                socket.close()
+                socket_.close()
```

Comment:
would the previous code just crash?

Reply:
Yes. `TypeError: close() missing 1 required positional argument: 'self'`